hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -732,20 +733,33 @@ private boolean o3Exists(final Path f) throws IOException {
 
   @Override
   public FileStatus[] listStatus(Path f) throws IOException {
+    return convertFileStatusArr(listStatusAdapter(f));
+  }
+
+  private FileStatus[] convertFileStatusArr(FileStatusAdapter[] adapterArr) {
+    FileStatus[] fileStatuses = new FileStatus[adapterArr.length];
+    int index = 0;
+    for (FileStatusAdapter statusAdapter : adapterArr) {
+      fileStatuses[index++] = convertFileStatus(statusAdapter);      
+    }
+    return fileStatuses;
+  }
+
+  
+  public FileStatusAdapter[] listStatusAdapter(Path f) throws IOException {
```

Comment:
I think we should rewrite it to return List<FileStatusAdapter>.
There's no point in creating a temp array that is going to be discarded right away.

Reply:
Yes, will do!